src/OrbitQt/ProfilingTargetDialog.cpp

```diff
@@ -458,32 +459,45 @@ void ProfilingTargetDialog::OnProcessListUpdate(
     process_model_.SetProcesses(process_list);
     emit ProcessListUpdated();
 
-    // In case there is a selection already, do not change anything
-    if (ui_->processesTableView->selectionModel()->hasSelection()) return;
+    // In case there is a selection already, do not change anything, only update the cpu usage
+    if (ui_->processesTableView->selectionModel()->hasSelection()) {
+      const auto it =
+          std::find_if(process_list.begin(), process_list.end(), [&](const auto& process) {
+            return process.pid() == ui_->processesTableView->selectionModel()
+                                        ->currentIndex()
+                                        .data(Qt::UserRole)
+                                        .value<const ProcessInfo*>()
+                                        ->pid();
```

Comment:
The right hand side of this comparison is constant, isn't it? It's hard to understand at first that this doesn't change. I'd suggest to put this pid into a variable first, and then search for the correct process by comparing to this variable.

Reply:
One more thing: The "current index" is somehow independent from a selection, and mainly meant for keyboard navigation.

You can have a selection and a different current index, or maybe no current index at all. I think you wanna ask the selection model for the selectedRows() and then take the first one.

Reply:
done